### DISCUSSION TITLE
HyperlinkButton in Avalonia

### DISCUSSION THREAD

**Discussion by ShrutiJaiswal1494** [AvaloniaUI/Avalonia#6664] [Q&A] [answered]
Hi,

I am looking for an alternative to Hyperlink Button (present in WPF). Is there a way to have a button click redirected to an URL? 
Any help is much appreciated. Thanks in advance.

**Comment by maxkatz6** [comment, marked as answer]
1. You can retemplate Button with a style to look like a link. It would be relatively simple style.
2. You can open link in whatever way you want with Click handler or Command. For instance https://github.com/dotnet/runtime/issues/17938

**Comment by nref** [comment]
Providing an example template. The text is underlined and highlights on hover. Your system accent colors will differ.

```xml
  <Style Selector="Button.hyperlink">
    <Setter Property="Template">
      <ControlTemplate>
        <TextBlock Text="{TemplateBinding Content}" Foreground="{StaticResource SystemAccentColor}" TextDecorations="Underline">
          <TextBlock.Styles>
            <Style Selector="TextBlock:pointerover">
              <Setter Property="Foreground" Value="{StaticResource SystemAccentColorLight1}"/>
            </Style>
          </TextBlock.Styles>
        </TextBlock>
      </ControlTemplate>
    </Setter>
    <Setter Property="Background" Value="Transparent"/>
    <Setter Property="BorderThickness" Value="0"/>
  </Style>

```

Example use

```xml
<Button Command="{Binding HandleLinkClicked}" Classes="hyperlink" Content="Open Link"/>
```

```csharp
  // Browser integration from Avalonia.Essentials
  public async Task HandleLinkClicked() => await Browser.OpenAsync("https://avaloniaui.net/");
```

![2023-08-24 link button](https://github.com/AvaloniaUI/Avalonia/assets/2089725/727e36c7-8430-4799-95de-c0d7aa69b71f)

**Reply by timunie** [reply]
❤️ thank you 🙏
